src/objects.rs

```diff
@@ -1979,6 +1981,33 @@ impl AddPrefix for Occupancy {
     }
 }
 
+impl Hash for Occupancy {
```

Comment:
Same here, looks weird. Maybe, if you want to disambiguate the "key" part of the Occupancy (where it applies: line, date, time, etc.), from the "value" part (how much occupancy?), you can change a little bit the `struct`.

```rust
#[derive(Debug, Default, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
pub struct OccupancyWindow {
    pub line_id: String,
    pub from_stop_area: String,
    pub to_stop_area: String,
    #[serde(
        deserialize_with = "de_from_date_string",
        serialize_with = "ser_from_naive_date"
    )]
    pub from_date: Date,
    #[serde(
        deserialize_with = "de_from_date_string",
        serialize_with = "ser_from_naive_date"
    )]
    pub to_date: Date,
    pub from_time: Time,
    pub to_time: Time,
    #[serde(
        deserialize_with = "de_opt_bool_from_str",
        serialize_with = "ser_from_opt_bool"
    )]
    pub monday: Option<bool>,
    #[serde(
        deserialize_with = "de_opt_bool_from_str",
        serialize_with = "ser_from_opt_bool"
    )]
    pub tuesday: Option<bool>,
    #[serde(
        deserialize_with = "de_opt_bool_from_str",
        serialize_with = "ser_from_opt_bool"
    )]
    pub wednesday: Option<bool>,
    #[serde(
        deserialize_with = "de_opt_bool_from_str",
        serialize_with = "ser_from_opt_bool"
    )]
    pub thursday: Option<bool>,
    #[serde(
        deserialize_with = "de_opt_bool_from_str",
        serialize_with = "ser_from_opt_bool"
    )]
    pub friday: Option<bool>,
    #[serde(
        deserialize_with = "de_opt_bool_from_str",
@@ -1967,6 +1968,7 @@
        serialize_with = "ser_from_opt_bool"
    )]
    pub sunday: Option<bool>,
}

#[derive(Debug, Default, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
pub struct Occupancy {
    #[serde(flatten)]
    pub occupancy_window: OccupancyWindow,
    pub occupancy: OccupancyStatus,
}
```

This way, you can compare `occupancy.occupancy_window` for deduplication? You can even store `HashMap<OccupancyWindow, OccupancyStatus>` :shrug: